framework/assets/yii.validation.js

```diff
@@ -236,6 +236,23 @@ yii.validation = (function ($) {
             }
         },
 
+        _trim: function(value, options = {skipOnEmpty: true}) {
+            if (options.skipOnEmpty && pub.isEmpty(value)) {
+                return value;
+            }
+  
+            value = new String(value);
+            if (options.chars || !String.prototype.trim) {
+                var chars = !options.chars
+                    ? ' \\s\xA0'
+                    : options.chars.replace(/([\[\]\(\)\.\?\/\*\{\}\+\$\^\:])/g, '\$1');
```

Comment:
The regexp was different in jQuery: https://github.com/jquery/jquery/commit/56e73e0c4ae908c853c986aadeda7cfcf2df4f6f. If we want it compatible, maybe it is a good idea to use that code.

Reply:
I guess, @arogachev meant it in https://github.com/yiisoft/yii2/pull/19254#discussion_r810956509

Reply:
Yes. And the regex is the same in offered polyfill here https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/String/Trim.

Reply:
@samdark @arogachev but it's not `$.trim()`/`String.prototype.trim()` polyfill - it's PHP `trim()` polyfill! look to `options.chars`.

Reply:
Why? You are replacing jQuery's `$.trim()` with this code so it's jQuery's trim for certain. Either that or this pull request does more than what is said in the title...

Reply:
@samdark 
> Replace deprecated $.trim() in yii.validation.js

> Adds optional option chars to emulate PHP trim()

where "more than what is said"?